sound/soc/sof/intel/hda.c

```diff
@@ -94,7 +95,7 @@ int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w)
 	return 0;
 }
 
-int hda_ctrl_dai_widget_free(struct snd_soc_dapm_widget *w)
+int hda_ctrl_dai_widget_free(struct snd_soc_dapm_widget *w, u8 flags)
```

Comment:
I'd not add the flags to free at all. We essentially have too IPCs now, DAI_CONFIG and DAI_CONFIG_FREE, and this new flag only applies to the non-free DAI_CONFIG we send when setting up a widget.

Reply:
right but I think its better to add the flags to the free also. Who knows whats coming in the future?